packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -213,4 +215,34 @@ export class AbapServiceProvider extends ServiceProvider {
 
         return this.services[subclassName] as T;
     }
+
+    /**
+     * Create a UI Service generator for the given business object.
+     *
+     * @param bo - business object
+     * @returns a UI Service generator
+     */
+    public async getUiServiceGenerator(bo: BusinessObject): Promise<UiServiceGenerator> {
+        const generatorService = await this.getAdtService<GeneratorService>(GeneratorService);
+        if (!generatorService) {
+            throw new Error('Generators are not support on this system');
+        }
+        const config = await generatorService.getUIServiceGeneratorConfig(bo.name);
+        // quick and dirty
+        const path = config.link[0].href.split(config.id)[0] + config.id;
```

Comment:
Would it be easier to read to use substring?  

```
const endIndex = config.link[0].href.indexOf(config.id) + config.id.length
const path = config.link[0].href.substring(0, endIndex);
```

Also need to be careful with `undefined` value in config.link[0].href which might break the execution.


It is also better to extract this url converter to a separate function to make it easy when you want to change the implementation later.

Reply:
Thanks, I've used you're suggested code moved it to a function and added a check in it.